### PR TITLE
Make STUN run by default when embedded DERP is enabled

### PR DESCRIPTION
**Pull request by juanfont**
This commit also allows to set an external STUN server, while running the embedded DERP server (without embedded STUN)

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ const (
 	errUnsupportedLetsEncryptChallengeType = Error(
 		"unknown value for Lets Encrypt challenge type",
 	)
+	errSTUNAddressNotSet = Error("STUN address not set")
```
Comment (kradalby) [inline app.go:65]: Define this at the top as a `const`

File: app.go [hunk 2]
```diff
@@ -500,6 +501,10 @@ func (h *Headscale) Serve() error {
 	h.DERPMap = GetDERPMap(h.cfg.DERP)
 
 	if h.cfg.DERP.ServerEnabled {
+		if h.cfg.DERP.STUNAddr == "" { // When embedded DERP is enabled we always need a STUN server address, embedded or external
```
Comment (kradalby) [inline app.go:504]:
```suggestion
		// When embedded DERP is enabled we always need a STUN server address, embedded or external
		if h.cfg.DERP.STUNAddr == "" {
```

File: config-example.yaml [hunk 3]
```diff
@@ -69,10 +69,13 @@ derp:
     region_code: "headscale"
     region_name: "Headscale Embedded DERP"
 
-    # If enabled, also listens in UDP at the configured address for STUN connections to help on NAT traversal
+    # Enabled by default when embedded DERP is enabled. Listens in UDP at the configured address for STUN connections
+    # to help on NAT traversal.
+    # If DERP is enabled, but STUN is disabled you still need to input an external STUN server in the listen_addr field.
```
Comment (kradalby) [inline config-example.yaml:74]: hmm, I am not sure about that name/config option being reused for the server if it is disabled, that sounds confusing :/...

File: app.go (hunk 1 above)
Comment (juanfont) [reply app.go:65]: ? Is a const at the top

File: config-example.yaml (hunk 3 above)
Comment (kradalby) [reply config-example.yaml:74]: how about `derp.override_stun_server`

**Review by kradalby — APPROVED** [review]

**Review by kradalby — APPROVED** [review]

**Pull request merged**